<launch>

    <arg name="pos_max" default="1"/>
    <arg name="orient_max" default="0.5" />
    <arg name="min_loop_freq" value="15" />

    <node name="controller" pkg="pid" type="controller" ns="global_x" output="screen" >
         <param name="node_name" value="global_x_pid" />
         <param name="Kp" value="0.5" />
         <param name="Ki" value="0.0" />
         <param name="Kd" value="0.07" />
         <param name="upper_limit" value="$(arg pos_max)" />
         <param name="lower_limit" value="-$(arg pos_max)" />
         <!-- <param name="windup_limit" value="10" /> -->
         <param name="min_loop_frequency" value="$(arg min_loop_freq)" />
         <remap from="control_effort" to="/control_effort/x" />
         <remap from="state" to="/state/x" />
    </node>

    <node name="controller" pkg="pid" type="controller" ns="global_y" output="screen" >
         <param name="node_name" value="global_y_pid" />
         <param name="Kp" value="0.5" />
         <param name="Ki" value="0.0" />
         <param name="Kd" value="0.04" />
         <param name="upper_limit" value="$(arg pos_max)" />
         <param name="lower_limit" value="-$(arg pos_max)" />
         <!-- <param name="windup_limit" value="10" /> -->
         <param name="min_loop_frequency" value="$(arg min_loop_freq)" />
         <remap from="control_effort" to="/control_effort/y" />
         <remap from="state" to="/state/y" />
    </node>

    <node name="controller" pkg="pid" type="controller" ns="global_z" output="screen" >
         <param name="node_name" value="global_z_pid" />
         <param name="Kp" value="0.9" />
         <param name="Ki" value="0.05" />
         <param name="Kd" value="0.01" />
         <param name="upper_limit" value="$(arg pos_max)" />
         <param name="lower_limit" value="-$(arg pos_max)" />
         <!-- <param name="windup_limit" value="10" /> -->
         <param name="min_loop_frequency" value="$(arg min_loop_freq)" />
         <remap from="control_effort" to="/control_effort/z" />
         <remap from="state" to="/state/z" />
    </node>

    <node name="controller" pkg="pid" type="controller" ns="global_roll" output="screen" >
         <param name="node_name" value="global_roll_pid" />
         <param name="Kp" value=".1" />
         <param name="Ki" value="0.0" />
         <param name="Kd" value="0.0" />
         <param name="upper_limit" value="$(arg orient_max)" />
         <param name="lower_limit" value="-$(arg orient_max)" />
         <!-- <param name="windup_limit" value="10" /> -->
         <param name="min_loop_frequency" value="$(arg min_loop_freq)" />
         <remap from="control_effort" to="/control_effort/roll" />
         <remap from="state" to="/state/roll" />
    </node>

    <node name="controller" pkg="pid" type="controller" ns="global_pitch" output="screen" >
         <param name="node_name" value="global_pitch_pid" />
         <param name="Kp" value="0.1" />
         <param name="Ki" value="0.0" />
         <param name="Kd" value="0.0" />
         <param name="upper_limit" value="$(arg orient_max)" />
         <param name="lower_limit" value="-$(arg orient_max)" />
         <!-- <param name="windup_limit" value="10" /> -->
         <param name="min_loop_frequency" value="$(arg min_loop_freq)" />
         <remap from="control_effort" to="/control_effort/pitch" />
         <remap from="state" to="/state/pitch" />
    </node>

    <node name="controller" pkg="pid" type="controller" ns="global_yaw" output="screen" >
         <param name="node_name" value="global_yaw_pid" />
         <param name="Kp" value="0.3" />
         <param name="Ki" value="0.04" />
         <param name="Kd" value="0.09" />
         <param name="upper_limit" value="$(arg orient_max)" />
         <param name="lower_limit" value="-$(arg orient_max)" />
         <!-- <param name="windup_limit" value="10" /> -->
         <param name="min_loop_frequency" value="$(arg min_loop_freq)" />
         <remap from="control_effort" to="/control_effort/yaw" />
         <remap from="state" to="/state/yaw" />
    </node>

</launch>
